crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -135,12 +137,207 @@ impl Persistence {
+                    .is_some()
+                {
+                    let quote = database::orders::read_quote(&mut ex, &ByteArray(order.0))
+                        .await
+                        .context("fetch quote")?
+                        .map(dto::quote::into_domain)
+                        .transpose()
+                        .map_err(|_| error::Auction::DbConversion("quote overflow"))?
+                        .ok_or(error::Auction::MissingQuote)?;
+
+                    let policies =
+                        database::fee_policies::fetch(&mut ex, auction_id, ByteArray(order.0))
+                            .await
+                            .context("fetch fee policies")?;
+
+                    for policy in policies {
+                        let policy = dto::fee_policy::into_domain(policy, &quote)?;
+                        fee_policies.entry(order).and_modify(|policies| {
+                            policies.push(policy);
+                        });
+                    }
+                }
+            }
+            fee_policies
+        };
+
+        Ok(domain::settlement::Auction {
+            id: auction_id,
+            prices,
+            deadline,
+            fee_policies,
+        })
+    }
+
+    /// Get winning competition solution.
+    pub async fn get_competition_solution(
+        &self,
+        auction_id: domain::auction::Id,
+    ) -> Result<domain::competition::Solution, error::Solution> {
+        let mut ex = self.postgres.pool.begin().await.context("begin")?;
+
+        let (winner, score) = {
+            let scores = database::settlement_scores::fetch(&mut ex, auction_id)
+                .await
+                .context("fetch scores")?
+                // if score is missing, no competition / auction exist for this auction_id
+                .ok_or(error::Solution::MissingScore)?;
+
+            (
+                H160(scores.winner.0).into(),
+                competition::Score::new(
+                    big_decimal_to_u256(&scores.winning_score)
+                        .ok_or(error::Solution::DbConversion("score"))?
+                        .into(),
+                )?,
+            )
+        };
+
+        // TODO: stabilize the solver competition table to get promised solution.
+        // https://github.com/cowprotocol/services/issues/2716
+        // for now, this is a `good enough` solution to unblock other tasks.
+        let solver_competition = database::solver_competition::load_by_id(&mut ex, auction_id)
+            .await
+            .context("load solver competition")?
+            .ok_or(error::Solution::SolverCompetition(anyhow::anyhow!(
+                "missing solver competition"
+            )))?;
+        let competition: model::solver_competition::SolverCompetitionDB =
+            serde_json::from_value(solver_competition.json).map_err(|_| {
+                error::Solution::SolverCompetition(anyhow::anyhow!("json conversion"))
```

Comment:
Ignoring the actual error in the error will make it impossible to debug this error.